src/box/lua/load_cfg.lua

```diff
@@ -248,6 +249,26 @@ returns. Set before first box.cfg{} call in order for the option to take effect.
 https://tarantool.io/compat/box_cfg_replication_sync_timeout
 ]]
 
+local replication_synchro_timeout_brief = [[
+Sets the default value for box.cfg.replication_synchro_timeout.
+Old is 5 seconds, new is infinity.
+
+https://tarantool.io/compat/box_cfg_replication_synchro_timeout
+]]
+
+local function replication_synchro_timeout_applicable(is_new)
+    if is_new and box_is_configured and
+        box.cfg.replication_synchro_timeout ~= 0 then
```

Comment:
Currently, `replication_synchro_timeout = 0` actually means, that it's 0, rollback immediately. This will cause downtime of the replicaset, when user wants to switch compat option (every new transaction will imediately rollback). We should fix that in `box_set_replication_synchro_timeout` and add a test, that transactions with 0 timeout don't immedately rollback.

Seems to me like it should be done in a separate commit

Reply:
Fixed.